pkg/unarchive/dmg.go

```diff
@@ -0,0 +1,126 @@
+package unarchive
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"path/filepath"
+
+	"github.com/schollz/progressbar/v3"
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/afero"
+)
+
+const FormatDMG string = "dmg"
+
+type dmgUnarchiver struct {
+	dest     string
+	executor Executor
+}
+
+type Executor interface {
+	HdiutilAttach(ctx context.Context, dmgPath, mountPoint string) (int, error)
+	HdiutilDetach(ctx context.Context, mountPath string) (int, error)
+}
+
+func cpFile(fs afero.Fs, src, dst string) error {
+	srcFile, err := fs.Open(src)
+	if err != nil {
+		return fmt.Errorf("open a file: %w", err)
+	}
+	defer srcFile.Close()
+
+	dstFile, err := fs.Create(dst)
```

Comment:
Maybe we should use the third party library such as https://github.com/otiai10/copy instead.
The current implementation is simple and maybe there are issues.
For example, the current implementation doesn't keep the file attributes such as permissions.
Of course we would be able to fix issues ourselves, but using the third party library we don't have to maintain the code.

But unfortunately, https://github.com/otiai10/copy doesn't support afero.

Reply:
Done. 9d6d87fcce192d5396f48e1e418356e347cb364e